### ISSUE TITLE
WhiteBox.setInternalState() does not always work when the argument is created with @Mock

### ISSUE THREAD

**Issue by johanhaleby** [jayway/powermock#200]
_From [mat...@gmail.com](https://code.google.com/u/105030434826960510651/) on October 08, 2009 13:57:38_

What steps will reproduce the problem? WhiteBox.setInternalState(Object, Object) does not work when the second
argument is created by using the @Mock annotation. However,
WhiteBox.setInternalState(Object, String, Object) works.

\########################################################################### What is the expected output? What do you see instead? The test should pass. 

org.powermock.reflect.exceptions.FieldNotFoundException: No field
assignable to "java.lang.Object" could be found in the class hierarchy of
SomeClass.

\########################################################################### What version of the product are you using? On what operating system? PowerMock 1.2.5

Ubuntu

\########################################################################### Please provide any additional information below. 5 files have been attached:
* InternalInterface.java is an interface with a method
* SomeClass.java is a class that has a private InternalInterface member
* FirstPassingTest.java is a junit test that creates an instance of
InternalInterface.java using the Mockito.mock() method and uses
setInternalState(Object, Object)
* SecondPassingTest.java is a junit test that creates an instance of 
InternalInterface.java using the @Mock annotation and uses
setInternalState(Object, String, Object)
* FailingTest.java is a junit test that creates an instance of
InternalInterface.java using the @Mock annotation and uses
setInternalState(Object, Object)

**Attachment:** [InternalInterface.java SomeClass.java FirstPassingTest.java SecondPassingTest.java FailingTest.java](http://code.google.com/p/powermock/issues/detail?id=180)

_Original issue: http://code.google.com/p/powermock/issues/detail?id=180_

**Comment by johanhaleby** [comment]
_From [johan.ha...@gmail.com](https://code.google.com/u/105676376875942041029/) on October 08, 2009 23:09:09_

Could you try PowerMock 1.3 and see if you get the same results then?

**Comment by johanhaleby** [comment]
_From [mat...@gmail.com](https://code.google.com/u/105030434826960510651/) on October 09, 2009 04:31:18_

Now the tests work, well done! I am sorry, I missed that you released the 1.3 version
the other day...

I also noticed that you have depricated
org.powermock.api.mockito.powermocklistener.AnnotationEnabler and
org.powermock.core.classloader.annotations.Mock. I retested the attached examples
without the @PowerMockListener and with the org.mockito.Mock instead, and the test
succeeds just as well. :-)
